ast/compile.go

```diff
@@ -853,6 +853,13 @@ func checkUndefinedFuncs(x interface{}, arity func(Ref) int) Errors {
 		if arity(ref) >= 0 {
 			return false
 		}
+		for i, r := range ref {
+			if v, ok := r.Value.(Var); ok {
+				if v0, ok := rwVars[v]; ok {
+					ref[i] = NewTerm(v0)
+				}
+			}
+		}
```

Comment:
nit: I wonder if we should just create a helper for this and return a copy of the original ref? Allocs are not a concern here since we're going to error out anyway.

Reply:
I'll look into it. We've got one already, I might reuse `rewriteVarsInRef` 😅 

I wonder if that thing needs to be that complicated, though, using TransformVars... Could there be vars in a Ref that don't satisfy
```go
ref[i].Value.(Var)
```
?

Reply:
Added another helper because the Transform-based one, unless I'm mistaken, specifically does its work in-place... also used the helper in another location where we had already been working with a copy anyways.